test/e2e/replica_set_tls/replica_set_tls_test.go

```diff
@@ -43,10 +43,10 @@ func TestReplicaSetTLS(t *testing.T) {
 
 	t.Run("Create MongoDB Resource", mongodbtests.CreateMongoDBResource(&mdb, ctx))
 	t.Run("Basic tests", mongodbtests.BasicFunctionality(&mdb))
-	t.Run("Wait for TLS to be enabled", tester.HasTlsMode("requireSSL", 60, WithTls()))
-	t.Run("Test Basic TLS Connectivity", tester.ConnectivitySucceeds(WithTls()))
-	t.Run("Test TLS required", tester.ConnectivityFails(WithoutTls()))
-	t.Run("Ensure Authentication", tester.EnsureAuthenticationIsConfigured(3, WithTls()))
+	mongodbtests.SkipTestIfLocal(t, "Wait for TLS to be enabled", tester.HasTlsMode("requireSSL", 60, WithTls()))
+	mongodbtests.SkipTestIfLocal(t, "Test Basic TLS Connectivity", tester.ConnectivitySucceeds(WithTls()))
+	mongodbtests.SkipTestIfLocal(t, "Test TLS required", tester.ConnectivityFails(WithoutTls()))
+	mongodbtests.SkipTestIfLocal(t, "Ensure Authentication", tester.EnsureAuthenticationIsConfigured(3, WithTls()))
```

Comment:
nice! We could also wrap these section like 

```golang
	mongodbtests.SkipTestIfLocal(t, "Ensure MongoDB TLS Configuration", func(t *testing.T) {
		tester.HasTlsMode("requireSSL", 60, WithTls())
		tester.ConnectivitySucceeds(WithTls())
		tester.ConnectivityFails(WithoutTls())
		tester.EnsureAuthenticationIsConfigured(3, WithTls())
	})
```

To make it a little more concise, will leave it  to you!